### ISSUE TITLE
Runtime warning and NameError

### ISSUE THREAD

**Issue by muppetjones** [zstephens/neat-genreads#4]
Just came across the following error in the `master` branch. I haven't had a chance to track it down, but I'll update if I do.

### Command

```shell
for i in $(seq 1 100)
do
  python genReads.py \
    -r /refs/ensembl/ensembl_build_75.fa \
    -R 126 \
    -o /data/neat/normal \
    --pe 300 30 \
    -t /data/regions.bed \
    -c 100 \
    -M 0 \
    --bam --vcf \
    --job $i 100
done
```

### Error

A single command is giving the following error:

```
Using default sequencing error model.
Warning: Read length of error model (101) does not match -R value (126), rescaling model...
Using default gc-bias model.
Using artificial fragment length distribution. mean=300, std=30
found index /refs/ensembl/ensembl_build_75.fa.fai
enumerating all non-N regions in reference sequence...
reading 1... 33.343 (sec)
/neat-genreads/py/probability.py:86: RuntimeWarning: divide by zero encountered in log
  w_range = [np.exp(k*np.log(l) - l - logFactorial_list[k]) for k in k_range]
/neat-genreads/py/probability.py:86: RuntimeWarning: invalid value encountered in double_scalars
  w_range = [np.exp(k*np.log(l) - l - logFactorial_list[k]) for k in k_range]
Traceback (most recent call last):
  File "/neat-genreads/genReads.py", line 622, in <module>
    main()
  File "/neat-genreads/genReads.py", line 497, in main
    sequences = SequenceContainer(start,refSequence[start:end],PLOIDS,overlap,READLEN,[MUT_MODEL]*PLOIDS,MUT_RATE,coverage_dat,onlyVCF=ONLY_VCF)
  File "/neat-genreads/py/SequenceContainer.py", line 33, in __init__
    self.indelsToAdd = [n.sample() for n in self.ind_pois]
  File "/neat-genreads/py/probability.py", line 63, in sample
    return degenerateVal
NameError: global name 'degenerateVal' is not defined
```

**Comment by muppetjones** [comment]
Quick follow up:

The `RuntimeWarning`s are pretty deep in the model generation, but it looks like there's an issue when generating both the `ind_pois` and `snp_pois`.

The `NameError` is valid: It is passed into `DiscreteDistribution` and saved as `self.degenerate`. The error crops up in `sample`, where `degenerateVal` is not defined.

```python
class DiscreteDistribution:
	def __init__(self, weights, values, degenerateVal=None, method='bisect'):
		...
		self.degenerate = degenerateVal
	
	def sample(self):
		if self.degenerate != None:
			return degenerateVal
		...
```

**Comment by zstephens** [comment]
Hi there! Thanks for the report.

I was able to quickly replicate this error, and I believe I fixed it. I just pushed the changes to the repository, let me know if you have any future issues!

Thanks!
Zach

On Oct 11, 2016, at 4:23 PM, Stephen J Bush <notifications@github.com<mailto:notifications@github.com>> wrote:


Just came across the following error in the master branch. I haven't had a chance to track it down, but I'll update if I do.

Command

for i in $(seq 1 100)
do
  python genReads.py \
    -r /refs/ensembl/ensembl_build_75.fa \
    -R 126 \
    -o /data/neat/normal \
    --pe 300 30 \
    -t /data/regions.bed \
    -c 100 \
    -M 0 \
    --bam --vcf \
    --job $i 100
done

Error

A single command is giving the following error:

Using default sequencing error model.
Warning: Read length of error model (101) does not match -R value (126), rescaling model...
Using default gc-bias model.
Using artificial fragment length distribution. mean=300, std=30
found index /refs/ensembl/ensembl_build_75.fa.fai
enumerating all non-N regions in reference sequence...
reading 1... 33.343 (sec)
/neat-genreads/py/probability.py:86: RuntimeWarning: divide by zero encountered in log
  w_range = [np.exp(k*np.log(l) - l - logFactorial_list[k]) for k in k_range]
/neat-genreads/py/probability.py:86: RuntimeWarning: invalid value encountered in double_scalars
  w_range = [np.exp(k*np.log(l) - l - logFactorial_list[k]) for k in k_range]
Traceback (most recent call last):
  File "/neat-genreads/genReads.py", line 622, in <module>
    main()
  File "/neat-genreads/genReads.py", line 497, in main
    sequences = SequenceContainer(start,refSequence[start:end],PLOIDS,overlap,READLEN,[MUT_MODEL]*PLOIDS,MUT_RATE,coverage_dat,onlyVCF=ONLY_VCF)
  File "/neat-genreads/py/SequenceContainer.py", line 33, in __init__
    self.indelsToAdd = [n.sample() for n in self.ind_pois]
  File "/neat-genreads/py/probability.py", line 63, in sample
    return degenerateVal
NameError: global name 'degenerateVal' is not defined
